core/services/ccip/orm.go

```diff
@@ -0,0 +1,169 @@
+package ccip
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/sqlutil"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/assets"
+)
+
+type GasPrice struct {
+	SourceChainSelector uint64
+	GasPrice            *assets.Wei
+	CreatedAt           time.Time
+}
+
+type GasPriceUpdate struct {
+	SourceChainSelector uint64
+	GasPrice            *assets.Wei
+}
+
+type TokenPrice struct {
+	TokenAddr  string
+	TokenPrice *assets.Wei
+	CreatedAt  time.Time
+}
+
+type TokenPriceUpdate struct {
+	TokenAddr  string
+	TokenPrice *assets.Wei
+}
+
+//go:generate mockery --quiet --name ORM --output ./mocks/ --case=underscore
+type ORM interface {
+	GetGasPricesByDestChain(ctx context.Context, destChainSelector uint64) ([]GasPrice, error)
+	GetTokenPricesByDestChain(ctx context.Context, destChainSelector uint64) ([]TokenPrice, error)
+
+	InsertGasPricesForDestChain(ctx context.Context, destChainSelector uint64, jobId int32, gasPrices []GasPriceUpdate) error
+	InsertTokenPricesForDestChain(ctx context.Context, destChainSelector uint64, jobId int32, tokenPrices []TokenPriceUpdate) error
+
+	ClearGasPricesByDestChain(ctx context.Context, destChainSelector uint64, to time.Time) error
+	ClearTokenPricesByDestChain(ctx context.Context, destChainSelector uint64, to time.Time) error
+}
+
+type orm struct {
+	ds sqlutil.DataSource
+}
+
+var _ ORM = (*orm)(nil)
+
+const (
+	gasTableName   = "ccip.observed_gas_prices"
+	tokenTableName = "ccip.observed_token_prices"
```

Comment:
Let's inline these table names. Using Sprintf to construct queries is risky. I'm surprised the linter isnt complaining.